### PR TITLE
Adds support for tectonic_etcd_scheme

### PR DESCRIPTION
**Pull request by coresolve**
This adds support for specifying the scheme that will be used for the
external etcd cluster. it retains the default `https`

I have tested bare metal azure and aws. @squat @kyoto

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -184,6 +184,15 @@ EOF
   default = []
 }
 
+variable "tectonic_etcd_scheme" {
+  type    = "string"
+  default = "https"
+
+  description = <<EOF
+(optional) Can be either "http" or "https" When set this scheme will be used for all provided etcd endpoints.
```
Comment (squat) [inline config.tf]: Please add a “.” between the two sentences and add a “,” after “When set” so it reads better.

File: platforms/gcp/main.tf [hunk 2]
```diff
@@ -61,6 +61,8 @@ module "etcd" {
   master_subnetwork_name = "${module.network.master_subnetwork_name}"
   external_endpoints     = ["${compact(var.tectonic_etcd_servers)}"]
 
+  etcd_scheme = "${var.tectonic_etcd_scheme}"
```
Comment (squat) [inline platforms/gcp/main.tf]: We need to declare “etcd” in platforms/gcp/etcd otherwise GCP will fail to build.

File: platforms/vmware/main.tf [hunk 3]
```diff
@@ -7,6 +7,7 @@ module "etcd" {
   container_image    = "${var.tectonic_container_images["etcd"]}"
   base_domain        = "${var.tectonic_base_domain}"
   external_endpoints = ["${compact(var.tectonic_etcd_servers)}"]
+  etcd_scheme        = "${var.tectonic_etcd_scheme}"
```
Comment (squat) [inline platforms/vmware/main.tf]: Same here, let’s declare etcd in platforms/vmware/etcd so as to not break this platform.

**Comment by mxinden** [comment]
We did some changes (https://github.com/coreos/tectonic-installer/pull/2082) to the testing process. Please rebase on to current master, so that the `basic-tests` PR status is reported correctly.

**Comment by coresolve** [comment]
We can't infer https only from certs. There are a few ways to configure etcd.
1. etcd non tls.
2. etcd tls but no client certs and no ca crt.
    This means that we the apiservers will be able to trust the etcd endpoints with the ca cert bundle they already have (/usr/share/ca-certificates)
3. etcd tls with etcd ca crt but not mtls
    This means the etcd cluster is signed by a CA that isn't represented in /usr/share/ca-certificates
    For example an Corp CA.
4. etcd with mtls
    This means that we provide client cert, key and optionally ca cert. In this case it's expected that the etcd servers are configured for mtls (and will assert that the client is signed by a cert that it knows)

File: platforms/gcp/main.tf (hunk 2 above)
Comment (coresolve) [reply platforms/gcp/main.tf]: fixed

File: platforms/vmware/main.tf (hunk 3 above)
Comment (coresolve) [reply platforms/vmware/main.tf]: fixed

**Comment by coresolve** [comment]
closed in favor of https://github.com/coreos/tectonic-installer/pull/2288

**Pull request closed without merging**